olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/StreamLayerClient.h

```diff
@@ -199,6 +199,34 @@ class DATASERVICE_READ_API StreamLayerClient final {
   client::CancellableFuture<DataResponse> GetData(
       const model::Message& message);
 
+  /**
+   * @brief Read messages from stream layer and commits the successfully
+   * consumed messaged before handing them over to the user
+   *
+   * Only possible if Subscribe() was called before.
+   * If the payload is more the 1MB then it is not embedded into the Metadata
+   * and the user needs to call GetData(Message) to actually download the data.
+   *
+   * @param callback The `PollResponseCallback` object that is invoked when the
+   * get data request is completed.
+   *
+   * @return A token that can be used to cancel this request.
+   */
+  client::CancellationToken Poll(PollResponseCallback callback);
+
+  /**
+   * @brief Read messages from stream layer and commits the successfully
+   * consumed messaged before handing them over to the user
+   *
+   * Only possible if Subscribe() was called before.
```

Comment:
```suggestion
   * Only possible if subscribed successfully.
```
Would that read more straightforward?